pkg/operator/ceph/admission.go

```diff
@@ -74,7 +76,7 @@ func ValidateCephResource(request *v1beta1.AdmissionRequest, context *clusterd.C
 	if isCephClusterCreate(request) {
 		//If external mode enabled, then check if other fields are empty
 		if reqCephCluster.Spec.External.Enable {
-			if reqCephCluster.Spec.Mon != (cephv1.MonSpec{}) || reqCephCluster.Spec.Dashboard != (cephv1.DashboardSpec{}) || reqCephCluster.Spec.Monitoring != (cephv1.MonitoringSpec{}) || reqCephCluster.Spec.DisruptionManagement != (cephv1.DisruptionManagementSpec{}) || len(reqCephCluster.Spec.Mgr.Modules) > 0 || len(reqCephCluster.Spec.Network.Provider) > 0 || len(reqCephCluster.Spec.Network.Selectors) > 0 {
+			if reqCephCluster.Spec.Mon != (cephv1.MonSpec{}) || reqCephCluster.Spec.Dashboard != (cephv1.DashboardSpec{}) || !reflect.DeepEqual(reqCephCluster.Spec.Monitoring, (cephv1.MonitoringSpec{})) || reqCephCluster.Spec.DisruptionManagement != (cephv1.DisruptionManagementSpec{}) || len(reqCephCluster.Spec.Mgr.Modules) > 0 || len(reqCephCluster.Spec.Network.Provider) > 0 || len(reqCephCluster.Spec.Network.Selectors) > 0 {
```

Comment:
Do we not need `DeepEqual()` checks against other types as well?

Reply:
We need DeepEqual() here because of `(struct containing []string cannot be compared)`, so for now the others are fine.